cpp/locality/AABBQuery.h

```diff
@@ -123,7 +123,20 @@ class AABBIterator : public NeighborQueryPerPointIterator
         : NeighborQueryPerPointIterator(neighbor_query, query_point, query_point_idx, r_max, r_min,
                                         exclude_ii),
           m_aabb_query(neighbor_query)
-    {}
+    {
```

Comment:
See previous comment -- revert this change and apply the validation logic in the `NeighborQueryPerPointIterator` [constructor](https://github.com/glotzerlab/freud/blob/eb2a97bac302ed45236f26e0a7e6a0346071c579/cpp/locality/NeighborQuery.h#L262-L268) instead. I think you can safely revert all the changes in this file and `LinkCell.h`.